fms/models/llama.py

```diff
@@ -531,22 +530,63 @@ def factory(**kwargs):
     _llama_factory_factory((_granite_8b_code_config)),
 )
 
+# Create all the pieces to generate adapters for different checkpoints
+serialization.register_adapter_step(
+    "llama", "pre0.0.6_unfused_to_fused", serialization._pre006_adapter_step
+)
+
+_unfused_to_fused = lambda sd, ea: serialization._mlp_glu_unfused_to_fused_adapter_step(
+    serialization._attn_unfused_to_fused_step(sd, ea), ea
+)
+
+
+def _weight_fusion(input_sd: Mapping, extra_kwargs: Optional[Mapping] = None):
+    has_fused_weights = True
+    if extra_kwargs and "model_config" in extra_kwargs:
+        if not extra_kwargs["model_config"]["fused_weights"]:
+            has_fused_weights = False
+
+    new_sd = input_sd
+    if has_fused_weights:
+        new_sd = _unfused_to_fused(new_sd, extra_kwargs)
+    return new_sd
+
+
+serialization.register_adapter_step("llama", "weight_fusion", _weight_fusion)
+
+
+def _hf_gptq_llama_check(input_sd: Mapping, extra_kwargs: Optional[Mapping] = None):
+    has_fused_weights = True
+    linear_type = "torch_linear"
+    if extra_kwargs and "model_config" in extra_kwargs:
+        if not extra_kwargs["model_config"]["fused_weights"]:
+            has_fused_weights = False
+        if extra_kwargs["model_config"]["linear_config"]:
+            linear_type = extra_kwargs["model_config"]["linear_config"]["linear_type"]
+
+    if "gptq" in linear_type and has_fused_weights:
+        raise ValueError(
+            "GPTQ HF llama checkpoints cannot be loaded into a model with fused weights"
```

Comment:
is this check not needed for `GPTBigCode`?

Reply:
no, GPTBigCode is fused by default in HF, so this doesn't apply to it